src/main/java/com/avispl/symphony/dal/device/core510i/QRCCommunicator.java

```diff
@@ -0,0 +1,436 @@
+/*
+ * Copyright (c) 2021 AVI-SPL, Inc. All Rights Reserved.
+ */
+package com.avispl.symphony.dal.device.core510i;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.net.Socket;
+import java.net.SocketTimeoutException;
+import java.util.Arrays;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import com.avispl.symphony.api.dal.dto.control.ConnectionState;
+import com.avispl.symphony.api.dal.error.CommandFailureException;
+import com.avispl.symphony.dal.BaseDevice;
+import com.avispl.symphony.dal.communicator.Communicator;
+import com.avispl.symphony.dal.communicator.ConnectionStatus;
+
+/**
+ * An implementation of QRCCommunicator to provide communication and interaction with QSys-Core
+ *
+ * @author Harry
+ * @since 1.0
+ */
```

Comment:
There are no specific requirement, but I think It would be better if you can add a break line between 30 and 31

Reply:
In the maven.yml file. I think you should add
- 'bugfix/**'
because we also need the build in bugfix branch to verify the bug

Reply:
Yes thank you, I fix it :D